.github/workflows/build-stitching.yml

```diff
@@ -0,0 +1,92 @@
+name: Build stitching binary
+on:
+  pull_request:
+    branches:
+      - develop
+      - main
+
+env:
+  THIRD_PARTY_DIR: "$GITHUB_WORKSPACE/3rdParty"
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, macos-10.15]
```

Comment:
I've tried adding `macos-10.15` without changing anything else, and I've immediately run into a whole host of [errors/warnings](https://github.com/jqmcginnis/build_biomedia_stitching/runs/8082494426?check_suite_focus=true#step:7:300) associated with the [`mia`](https://github.com/biomedia-mira/stitching/tree/master/mia) library distributed alongside `stitching`.

Many of the warnings reference C++11 extensions, so it might be worth following the advice of [this SO answer](https://stackoverflow.com/a/45291997) and compile with `-std=c++11` if possible. (Alternatively, specifying C++11 through CMake might be done via [these instructions instead[(https://stackoverflow.com/questions/10851247/how-do-i-activate-c-11-in-cmake).)

Reply:
Ah, now I'm definitely sure this is related to the compiler not using the C++11 standard. One of the errors is:

```
/Users/runner/work/build_biomedia_stitching/build_biomedia_stitching/mia/src/miaImage.cpp:68:21: error: expected expression
    , m_data( data, [](float*) {} )
                    ^
```

This `[](float*)` syntax is a [lambda expression](https://docs.microsoft.com/en-us/cpp/cpp/lambda-expressions-in-cpp?view=msvc-170), a feature that was [introduced in C++11](https://smartbear.com/blog/the-biggest-changes-in-c11-and-why-you-should-care/).

Reply:
Good news! Adding `set(CMAKE_CXX_STANDARD 11)` allowed `mia` to build error-free on macOS. 

(The error is due to `ldd` not being present on macOS, and has nothing to do with the actual build process.)

So, all that's left is Windows. :smile: